qiskit_experiments/library/characterization/fine_frequency.py

```diff
@@ -0,0 +1,130 @@
+    of the pulses do not match the qubit frequency then the qubit will precess in the
+    drive frame during the idle time and phase errors will build-up. By fitting the measured
+    points we can extract the error in the qubit frequency. The circuit that are run are
+
+    .. parsed-literal::
+
+                ┌────┐┌────────────────┐┌──────────┐┌────┐ ░ ┌─┐
+             q: ┤ √X ├┤ Delay(n*τ[dt]) ├┤ Rz(nπ/2) ├┤ √X ├─░─┤M├
+                └────┘└────────────────┘└──────────┘└────┘ ░ └╥┘
+        meas: 1/══════════════════════════════════════════════╩═
+                                                              0
+
+    """
+
+    def __init__(
+        self, qubit: int, backend: Optional[Backend] = None, repetitions: Optional[List[int]] = None
+    ):
+        """Setup a fine frequency experiment on the given qubit.
+
+        Args:
+            qubit: The qubit on which to run the fine frequency characterization experiment.
+            backend: Optional, the backend to run the experiment on.
+            repetitions: The number of repetitions, if not given then the default value
+                from the experiment default options will be used.
+        """
+        super().__init__([qubit], analysis=FineAmplitudeAnalysis(), backend=backend)
+
+        # Set default analysis options
+        self.analysis.set_options(angle_per_gate=np.pi / 2, phase_offset=0)
+
+        if repetitions is not None:
+            self.set_experiment_options(repetitions=repetitions)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine frequency experiment.
+
+        Experiment Options:
+            repetitions (List[int]): A list of the number of times that the identity is repeated.
+            sq_gate_duration (int): The duration of the single-qubit gate as the number of arbitrary
+                waveform generator samples it contains.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(40))
+        options.sq_gate_duration = None
+
+        return options
+
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        """Default transpiler options."""
+        options = super()._default_transpile_options()
+        options.inst_map = None
+        return options
+
+    def _pre_circuit(self) -> QuantumCircuit:
+        """A method that subclasses can override to perform gates before the main sequence."""
+        return QuantumCircuit(1)
+
+    def circuits(self) -> List[QuantumCircuit]:
+        """Return the list of quantum circuits to run."""
+
+        # Find out the duration of the sx gate from instructions map or the backend if missing.
+        if self.experiment_options.sq_gate_duration is None:
+            circuit = QuantumCircuit(1)
+            circuit.sx(0)
+            if self.transpile_options.inst_map is not None:
```

Comment:
Can this be a logic in `_set_backend` method? This checks an instmap in the transpile option, it uses the instmap otherwise take the map from the backend, finally raise an error saying no instmap (probably this is problematic to simulator). 

By the way why unit delay should be equivalent to single qubit gate duration? Seems like this is IBM-ish convention from old definition of Id gate (I don't think every quantum computer uses such definition). I prefer removing this complicated logic (and transpile/schedule is heavy computation overhead due to tons of class instantiation so must not be readily used) and just ask user to set delay in dt, which could be default to hard-coded 160 dt.

Reply:
This is now made more flexible here 81cfc40